Include/monitoring.h

```diff
@@ -0,0 +1,306 @@
+#ifndef Py_MONITORING_H
+#define Py_MONITORING_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct _PyMonitoringState {
+    uint8_t active;
+    uint8_t opaque;
+} PyMonitoringState;
+
+
+#  define Py_CPYTHON_MONITORING_H
+#  include "cpython/monitoring.h"
+#  undef Py_CPYTHON_MONITORING_H
+
+#define _PyMonitoring_IF_ACTIVE(STATE, X)  \
+    if ((STATE)->active) { \
+        return (X); \
+    } \
+    else { \
+        return 0; \
+    }
+
+
+#ifndef Py_LIMITED_API
+static inline void
+PyUnstable_Monitoring_EnterScope(PyMonitoringState *state_array, uint64_t *version,
```

Comment:
There's not much point in making this inline if it just calls another function.
The version test needs to be in the inline function.

Reply:
It's a macro.